generatebundlefile/promote.go

```diff
@@ -49,23 +58,48 @@ func GetSDKClients(profile string) (*SDKClients, error) {
 	if err != nil {
 		return nil, fmt.Errorf("Unable to create SDK connection to ECR %s", err)
 	}
+	return clients, nil
+}
 
-	if profile != "" {
-		confWithProfile, err := config.LoadDefaultConfig(context.TODO(),
-			config.WithRegion(ecrPublicRegion),
-			config.WithSharedConfigProfile(profile))
-		if err != nil {
-			return nil, fmt.Errorf("creating public AWS client config: %w", err)
-		}
+func (c *SDKClients) GetProfileSDKConnection(service, profile string) (*SDKClients, error) {
+	if service == "" || profile == "" {
+		return nil, fmt.Errorf("empty service, or profile passed to GetProfileSDKConnection")
+	}
+	var region = ecrRegion
+	if service == "ecrpublic" {
+		region = ecrPublicRegion
+	}
+	confWithProfile, err := config.LoadDefaultConfig(context.TODO(),
+		config.WithRegion(region),
+		config.WithSharedConfigProfile(profile))
+	if err != nil {
+		return nil, fmt.Errorf("creating public AWS client config: %w", err)
+	}
 
+	switch service {
```

Comment:
This is the switch that should be in NewECRClient and it would hide all this EcrPublicClient and EcrPrivateClient would have the same interface